docs/DeveloperGuide.md

```diff
@@ -5,48 +5,141 @@ title: Developer Guide
+- **User-friendliness**: With the benefits of having a Graphical User Interface (GUI), **users can easily navigate through contact information** to find exactly what they need to **connect** with others.
 
-This User Guide will provide you with an in-depth documentation for you to easily integrate ConnectUS into your daily life. It covers **step-by-step instructions** on setting up ConnectUS, core ConnectUS features and commands, and a glossary for definitions of terms used in ConnectUS.
+This Developer Guide provides an in-depth documentation on how ConnectUS is designed and implemented. It covers high-level details such as the architecture of ConnectUS, to detailed specifications on smaller pieces of the design such as how commands are implemented. It also includes a glossary for definitions of terms used in ConnectUS.
 
-So what are you waiting for? Get ready to **Connect** with others and let **US** handle the rest!
+You can use this guide to help maintain, upgrade, and evolve ConnectUS.
 
 ---
 
 <div style="page-break-after: always"></div>
 
-* Table of Contents
-{:toc}
+# Table of Contents
+
+- [1. Preface](#1-preface)
+  - [1.1 Acknowledgements](#11-acknowledgements)
+  - [1.2 Setting Up, Getting Started](#12-setting-up-getting-started)
+- [2. How to Use the Developer Guide](#2-how-to-use-the-developer-guide)
+  - [2.1 Notation](#21-notation)
+  - [2.2 Navigation](#22-navigation)
+- [3. Design](#3-design)
+  - [3.1 Architecture](#31-architecture)
+  - [3.2 UI Component](#32-ui-component)
+  - [3.3 Logic Component](#33-logic-component)
+  - [3.4 Model Component](#34-model-component)
+  - [3.5 Storage Component](#35-storage-component)
+  - [3.6 Common Classes](#36-common-classes)
+- [4. Implementation](#4-implementation)
+  - [4.1 Add Command](#41-add-command)
+  - [4.2 Edit Command](#42-edit-command)
+  - [4.3 Delete Command](#43-delete-command)
+  - [4.4 Help Command](#44-help-command)
+  - [4.5 Adding Additional Tags Command](#45-adding-additional-tags-command)
+  - [4.6 Deleting Individual Tags Command](#46-deleting-individual-tags-command)
+  - [4.7 Search Command](#47-search-command)
+  - [4.8 Upcoming Birthdays Command](#48-upcoming-birthdays-command)
+  - [4.9 Open Command](#49-open-command)
+- [5. Planned Enhancements](#5-planned-enhancements)
+  - [5.1 Improve Edit Command](#51-improve-edit-command)
+  - [5.2 Better Information Field Validation](#52-better-information-field-validation)
+  - [5.3 More Language Support](#53-more-language-support)
+  - [5.4 More Social Media Support](#54-more-social-media-support)
+  - [5.5 Improve UI](#55-improve-ui)
+  - [5.6 Improve Tag Deletion Command](#56-improve-tag-deletion-command)
+  - [5.7 Improve Consistency of Command Feedback](#57-improve-consistency-of-command-feedback)
+- [6.Documentation, Testing, and Other Guides ](#6-documentation-testing-and-other-guides)
```

Comment:
Maybe we can add a white space too (i.e. `6. Doc`) to make it consistent with other titles

Reply:
Thanks for pointing out!